### PR TITLE
Update search page to work with content API

### PR DESCRIPTION
**Pull request by pingzing**
A few bits of weirdness here. See inline comments.

Note: I updated the `find-<thing>` components, but didn't actually add them to the search page--I figured you had a plan here and didn't want to step on toes.

### CODE REVIEW

File: packages/client/src/app/pages/search/search.component.html [hunk 1]
```diff
@@ -1,5 +1,142 @@
...
\ No newline at end of file
+<div class="search-nav">
+    <div class="search-bar">
+        <form [formGroup]="searchForm" (ngSubmit)="submitSearch()">
+            <input formControlName="query" type="text" class="u-full-width" placeholder="Search ...">
+        </form>
+    </div>
+</div>
+
+<ng-container *ngIf="route.children.length === 0">
```
Comment (pingzing) [inline packages/client/src/app/pages/search/search.component.html:9]: Note about this page: all three search result types are very visually different. I did a little bit of cleanup here, but not much.

File: packages/client/src/app/pages/search/search.component.less [hunk 2]
```diff
@@ -19,9 +19,10 @@ div.search-nav {
             color: whitesmoke;
             box-shadow: none;
             font-size: 32px;
+            height: auto;
             border: 1px solid whitesmoke;
             margin-bottom: 0;
-            padding: 5px 20px 5px 20px;
+            padding: 10px 20px 10px 20px;
```
Comment (pingzing) [inline packages/client/src/app/pages/search/search.component.less:25]: Giving the search text a bit of room to breathe...

File: packages/client/src/app/pages/search/search.component.less [hunk 3]
```diff
@@ -217,6 +219,7 @@ div.search-results {
 
                     div.profile {
                         width: 100%;
+                        min-width: 0;
```
Comment (pingzing) [inline packages/client/src/app/pages/search/search.component.less:222]: Required for the username truncation to work

File: packages/client/src/app/pages/search/search.component.less [hunk 4]
```diff
@@ -226,6 +229,9 @@ div.search-results {
                                 font-weight: 500;
                                 font-size: 38px;
                                 margin-bottom: 0;
+                                white-space: nowrap;
+                                text-overflow: ellipsis;
+                                overflow: hidden;
```
Comment (pingzing) [inline packages/client/src/app/pages/search/search.component.less:234]: So stupid-long usernames don't overflow

File: packages/models/src/lib/util/initial-results.model.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { ContentModel } from '../content';
+import { User } from '../users';
+
+export interface InitialResults {
+    users: User[],
+    blogs: ContentModel[],
+    works: ContentModel[],
```
Comment (pingzing) [inline packages/models/src/lib/util/initial-results.model.ts:7]: This is one bit of weirdness--the initial implementation for searching just hits the content service, so it just gets back ContentModels. I could've implemented searches down in the individual services, but it doesn't look like we need that yet. (if we do in the future, we can always implement it there too)

File: packages/models/src/lib/util/paginate-result.model.ts [hunk 6]
```diff
@@ -2,13 +2,12 @@ export interface PaginateResult<T> {
...
-    readonly offset?: number;
-    readonly prevPage: number;
-    readonly nextPage: number;
+    readonly nextPage?: number;
+    readonly prevPage?: number;
     readonly pagingCounter: number;
-    readonly meta?: {};
+    readonly hasPrevPage: boolean;
+    readonly hasNextPage: boolean;
+    readonly meta?: any;
```
Comment (pingzing) [inline packages/models/src/lib/util/paginate-result.model.ts:12]: Some extremely minor cleanup to make this match this interface match the mongoose-side type so they're mutually-compatible. Now we don't have to rely on `any`-based duck-typing.

File: packages/server/src/app/db/users/users.service.ts [hunk 7]
```diff
@@ -307,28 +307,18 @@ export class UsersService {
...
-     * Finds the first six matches given the provided search parameters.
-     * For use with the initial page of search results.
-     * 
-     * @param query The relevant search parameters
-     */
-    async findInitialRelatedUsers(query: string): Promise<documents.UserDocument[]>{
-        return await this.userModel.find({$text: {$search: query}})
-            .select('-password -agreedToPolicies -audit.sessions -audit.termsAgree -audit.emailConfirmed -audit.isDeleted')
-            .limit(6);
-    }
```
Comment (pingzing) [inline packages/server/src/app/db/users/users.service.ts:320]: Folded this into `findRelatedUsers`.

**Review by Figments — APPROVED** [review]

**Pull request merged**